src/SystemManager.cc

```diff
@@ -460,58 +452,41 @@ void SystemManager::ProcessRemovedEntities(
     return;
   }
 
-  RemoveFromVectorIf(this->systemsReset,
-    [&](const SystemIfaceWithParent<ISystemReset>& system) {
-      return _ecm.IsMarkedForRemoval(system.parent);
-    });
-  RemoveFromVectorIf(this->systemsPreupdate,
-    [&](const SystemIfaceWithParent<ISystemPreUpdate>& system) {
-      return _ecm.IsMarkedForRemoval(system.parent);
-    });
-  RemoveFromVectorIf(this->systemsUpdate,
-    [&](const SystemIfaceWithParent<ISystemUpdate>& system) {
-      return _ecm.IsMarkedForRemoval(system.parent);
-    });
+  std::vector<SystemInternal> markedForRemoval;
+  for (const auto &system : this->systems)
+  {
+    if (_ecm.IsMarkedForRemoval(system.parentEntity))
+    {
+      markedForRemoval.push_back(system);
+    }
+  }
 
-  std::unordered_set<ISystemPostUpdate *> markedForRemoval;
-  for (std::size_t i = 0; i < this->systemsPostupdate.size(); i++)
+  for (const auto &system : markedForRemoval)
   {
-    if(_ecm.IsMarkedForRemoval(postUpdateParents[i]))
+    removeFromVector(this->systemsReset, system.reset);
+    removeFromVector(this->systemsPreupdate, system.preupdate);
+    removeFromVector(this->systemsUpdate, system.update);
+
+    if (system.postupdate)
     {
       // If system with a PostUpdate is marked for removal
       // mark all worker threads for removal.
       _needsCleanUp = true;
-      markedForRemoval.insert(this->systemsPostupdate[i]);
     }
-  }
 
-  RemoveFromVectorIf(this->systemsPostupdate,
-    [&](const auto system) {
-      if (markedForRemoval.count(system)) {
-        return true;
-      }
-      return false;
-    });
+    removeFromVector(this->systemsPostupdate, system.postupdate);
+    removeFromVector(this->systemsConfigure, system.configure);
+    removeFromVector(this->systemsConfigureParameters,
+                     system.configureParameters);
+    RemoveFromVectorIf(this->systems,
```

Comment:
This can be moved out of the loop using: `_ecm.IsMarkedForRemoval(system.parentEntity);`

Reply:
See #2475